appeals/web/src/server/appeals/appeal-details/appellant-case/outcome-incomplete/outcome-incomplete.controller.js

```diff
@@ -0,0 +1,230 @@
+	const appealReferenceFragments = appealReference.split('/');
+
+	return response.render('appeals/appeal/update-due-date.njk', {
+		appeal: {
+			id: appealId,
+			shortReference: appealReferenceFragments?.[appealReferenceFragments.length - 1]
+		},
+		page: {
+			title: 'Appellant case due date'
+		},
+		skipButtonUrl: `/appeals-service/appeal-details/${appealId}/appellant-case/check-your-answers`,
+		errors
+	});
+};
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ * @param {string} appealShortReference
+ * @param {number} appealId
+ */
+export const renderDecisionIncompleteConfirmationPage = async (
+	response,
+	appealShortReference,
+	appealId
+) => {
+	response.render('app/confirmation.njk', {
+		panel: {
+			title: 'Appeal incomplete',
+			appealReference: {
+				label: 'Appeal ID',
+				reference: appealShortReference
+			}
+		},
+		body: {
+			preTitle: 'The appeal has been reviewed.',
+			title: {
+				text: 'What happens next'
+			},
+			rows: [
+				{
+					text: 'We’ve sent an email to the appellant and LPA to inform the case is incomplete, and let them know what to do next.'
+				},
+				{
+					text: 'We also sent them a reminder about the appeal’s due date.'
+				},
+				{
+					text: 'Go to case details',
+					href: `/appeals-service/appeal-details/${appealId}`
+				}
+			]
+		}
+	});
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>}  */
+export const getIncompleteReason = async (request, response) => {
+	renderIncompleteReason(request, response);
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>} */
+export const postIncompleteReason = async (request, response) => {
+	const { errors } = request;
+
+	if (errors) {
+		return renderIncompleteReason(request, response);
+	}
+
+	try {
+		// @ts-ignore
+		request.session.appellantCaseReviewOutcome = mapPostedReviewOutcomeToApiReviewOutcome(
+			'incomplete',
+			request.body.incompleteReason,
+			request.body.otherReason
+		);
+
+		return response.redirect(
+			// @ts-ignore
+			`/appeals-service/appeal-details/${request.session.appealId}/appellant-case/incomplete/date`
+		);
+	} catch (error) {
+		logger.error(
+			error,
+			error instanceof Error
+				? error.message
+				: 'Something went wrong when completing appellant case review'
+		);
+
+		return response.render('app/500.njk');
+	}
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>}  */
+export const getUpdateDueDate = async (request, response) => {
+	renderUpdateDueDate(request, response);
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>} */
+export const postUpdateDueDate = async (request, response) => {
+	const { errors, body } = request;
+
+	if (errors) {
+		return renderUpdateDueDate(request, response);
+	}
+
+	if (!objectContainsAllKeys(request.session, 'appellantCaseReviewOutcome')) {
+		console.log("!objectContainsAllKeys(request.session, 'appellantCaseReviewOutcome')");
```

Comment:
We should be using pino for logging (if these are meant to be kept in)

Reply:
Ah, good spot, I left these in by mistake

Reply:
This has been removed now